### ISSUE TITLE
SSO with Azure AD

### ISSUE THREAD

**Issue by GoodFeel91** [derricksmith/phpsaml#39]
Hi,

I have GLPI 9.5 with the last version of PHPSAML plugin (1.1).

I created entreprise app on my Azure AD tenant and I follow the README but I have an error when I try to log with my Azure AD account (founded in glpi/files/_log/php-errors.log) :

```
[2021-02-02 10:55:13] glpiphplog.CRITICAL:   *** Uncaught Exception Error: Call to a member function checkUserData() on null in /var/www/html/glpi/plugins/phpsaml/inc/phpsaml.class.php at line 64
  Backtrace :
  .../www/html/glpi/plugins/phpsaml/front/acs.php:42 PluginPhpsamlPhpsaml::glpiLogin()
```

There is my PHPSAML configuration :

![image](https://user-images.githubusercontent.com/78422494/106602972-fe6f4100-655d-11eb-8527-01cba2f3f40e.png)

Thanks for help.

Best regards.

**Comment by derricksmith** [comment]
Does a user exist in GLPI with the email address you are logging into Microsoft SSO with?

**Comment by GoodFeel91** [comment]
Hi @derricksmith 

Thanks for your reply. I confirm if I create a user account with email adress such as ID on GLPI before sign-in with Azure AD SSO, it's works.

I provisionned users from my enterprise app Azure AD to connect to GLPI, but I need to provision GLPI too.

In my GLPI environment, my users was synced with my AD on-prem and the field "Mail address" is filled from my AD. 

I'm giving you another issue with Azure AD : PasswordLess connection doesn't works. I have MFA enabled on my Azure AD tenant and I use PasswordLess feature to sign-in. But when I test for GLPI, it fails with basic error (an error occurs). I was able to connect with SSO by using OTP (I use Microsoft Authenticator app).

Thanks for this plugin :)
